### PR TITLE
Added augmentations

### PR DESCRIPTION
**Pull request by pfabreu**
Sorry to pull to master 😳, it's an entirely self-contained folder though

### CODE REVIEW

File: arc/augmentations/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+Requires https://github.com/arc-community/arc installed
```
Comment (yk) [inline arc/augmentations/README.md]: probably not necessary anymore, right?

File: arc/augmentations/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+```
+See the demonstrations notebook for more details.
+
+Each augmentation class has a corresponding function in augmentations.functional that take in params to do a specific non random transformation, which is also useful for people that just want to use the same augmentation for all their data.
+
+
+You can sample params for a specific function in augmentations.functional by calling the Random* class's static method `get_params()` for example:
+
+```
+from augmentations.classes import RandomCropInputAndOuput
```
Comment (yk) [inline arc/augmentations/README.md]: is this now `arc.augmentations`?

File: arc/augmentations/augmentations/classes/color.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import functools
+import random
+from arc.interface import BoardPair,Riddle
+from typing import Union
+from augmentations import functional
+from augmentations.classes.helpers import same_aug_for_all_pairs_helper
```
Comment (yk) [inline arc/augmentations/augmentations/classes/color.py]: here also probably `arc.augmentations`

File: arc/augmentations/augmentations/classes/noise.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import functools
+import random
+import numpy as np
+from arc.interface import BoardPair, Riddle
+from typing import Union, Tuple
+from .. import functional
```
Comment (yk) [inline arc/augmentations/augmentations/classes/noise.py]: use full package name

File: arc/augmentations/augmentations/classes/noise.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+from augmentations.classes.helpers import same_aug_for_all_pairs_helper
```
Comment (yk) [inline arc/augmentations/augmentations/classes/noise.py]: also `arc.`

File: arc/augmentations/augmentations/classes/spatial.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
+import random
+from arc.interface import BoardPair, Riddle
+from typing import Union
+from .. import functional
+from ..functional import Direction
+from augmentations.classes.helpers import p_and_same_aug_helper, same_aug_for_all_pairs_helper
```
Comment (yk) [inline arc/augmentations/augmentations/classes/spatial.py]: same. I'll stop writing this now :D

**Review by yk — APPROVED** [review]
lgtm. package names should be adjusted

File: arc/augmentations/README.md (hunk 1 above)
Comment (pfabreu) [reply arc/augmentations/README.md]: yes

File: arc/augmentations/README.md (hunk 2 above)
Comment (pfabreu) [reply arc/augmentations/README.md]: yes, if we install the full arc then it should be the highest level, will fix it

File: arc/augmentations/augmentations/classes/noise.py (hunk 4 above)
Comment (pfabreu) [reply arc/augmentations/augmentations/classes/noise.py]: ok

**Comment by yk** [comment]
@pedro-abreu could you let pre-commit run over the code? then we're good

**Comment by pfabreu** [comment]
yeah, it was just getting stuck on a useless `if __name__ == '__main__'`

**Comment by yk** [comment]
I mean run `pre-commit run` to format / lint the code

**Pull request merged**